tests/phpunit/tests/user/wp-user/getUserBy.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/**
+ * @group  user
+ * @covers WP_User::get_data_by
+ */
+class Tests_User_WpUser_GetDataBy extends WP_UnitTestCase {
+	protected static $user_id;
+	protected static $user_data = array(
+		'user_login'    => 'user1',
+		'user_nicename' => 'userone',
+		'user_pass'     => 'password',
+		'first_name'    => 'John',
+		'last_name'     => 'Doe',
+		'display_name'  => 'John Doe',
+		'user_email'    => 'jonny@battlefield3.com',
+		'user_url'      => 'http://tacos.com',
+		'role'          => 'contributor',
+		'nickname'      => 'Johnny',
+		'description'   => 'I am a WordPress user that cares about privacy.',
+	);
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$user_id = $factory->user->create( self::$user_data );
+	}
+
+	/**
+	 * @dataProvider data_id_field_for_valid_userid
+	 */
+	public function test_id_field_for_valid_userid( $cast_func ) {
```

Comment:
```suggestion
	/**
	 * @dataProvider data_id_field_for_valid_userid
	 *
	 * @param string $cast_func Name of a function to cast the user ID to a different type.
	 */
	public function test_id_field_for_valid_userid( $cast_func ) {
```
Parameters from data providers should be documented.